coresdk/src/coresdk/json.cpp

```diff
@@ -245,11 +245,49 @@ namespace splashkit_lib
     
     bool json_has_key(json j, string key)
     {
+        if (INVALID_PTR(j, JSON_PTR))
+        {
+            LOG(ERROR) << "Invalid json object passed to json_has_key";
+            return false;
+        }
+
         return j->data.count(key) > 0;
     }
 
     int json_count_keys(json j)
     {
+        if (INVALID_PTR(j, JSON_PTR))
+        {
+            LOG(ERROR) << "Invalid json object passed to json_count_keys";
+            return 0;
+        }
+
         return static_cast<int>(j->data.size());
     }
+
+    json json_from_color(color clr)
+    {
+        json result = create_json();
+        json_add_number(result, "r", clr.r);
+        json_add_number(result, "g", clr.g);
+        json_add_number(result, "b", clr.b);
+        json_add_number(result, "a", clr.a);
+        return result;
+    }
+
+    color json_to_color(json j)
+    {
+        if (INVALID_PTR(j, JSON_PTR))
+        {
+            LOG(ERROR) << "Invalid json object passed to json_to_color";
+            return COLOR_WHITE;
+        }
+
+        color clr;
+        clr.r = json_read_number(j, "r");
+        clr.g = json_read_number(j, "g");
+        clr.b = json_read_number(j, "b");
+        clr.a = json_read_number(j, "a");
```

Comment:
@jarmstrong Rather than converting into components you could use `color_to_string` and `color_from_string` which generates a "#rrggbbaa" string.